torch/csrc/autograd/python_variable.cpp

```diff
@@ -504,6 +504,16 @@ PyObject *THPVariable_get_requires_grad(THPVariable *self, void *unused)
   END_HANDLE_TH_ERRORS
 }
 
+PyObject *THPVariable_retains_grad(THPVariable *self, void *unused)
+{
+  HANDLE_TH_ERRORS
+  if (check_has_torch_function((PyObject *)self)) {
+    return handle_torch_function_getter(self, "retains_grad");
+  }
+  return PyBool_FromLong(THPVariable_Unpack(self).retains_grad());
```

Comment:
It's surprising to convert form Long here given that retains_grad is already a bool.
You might want to check the bool in c++ and then use `Py_RETURN_TRUE` and `Py_RETURN_FALSE`.

Reply:
Yeah that is interesting, fixed.